main.js

```diff
@@ -0,0 +1,77 @@
+import { InstanceBase, InstanceStatus, runEntrypoint, TCPHelper } from '@companion-module/base'
+import { ConfigFields } from './config.js'
+import { getActionDefinitions } from './actions.js'
+import { getPresetDefinitions } from './presets.js'
+
+class GenericTcpUdpInstance extends InstanceBase {
+	async init(config) {
+		this.config = config
+
+
+		this.setActionDefinitions(getActionDefinitions(this))
+		this.setPresetDefinitions(getPresetDefinitions())
+		await this.configUpdated(config)
+	}
+
+	async configUpdated(config) {
+		if (this.socket) {
+			this.socket.destroy()
+			delete this.socket
+		}
+
+		this.config = config
+		this.init_tcp()
+	}
+
+	async destroy() {
+		if (this.socket) {
+			this.socket.destroy()
+		} else {
+			this.updateStatus(InstanceStatus.Disconnected)
+		}
+	}
+
+	getConfigFields() {
+		return ConfigFields
+	}
+
+	init_tcp() {
+		if (this.socket) {
+			this.socket.destroy()
+			delete this.socket
+		}
+
+		this.updateStatus(InstanceStatus.Connecting)
+
+		if (this.config.host) {
+			this.socket = new TCPHelper(this.config.host, this.config.port)
+
+			this.socket.on('status_change', (status, message) => {
+				this.updateStatus(status, message)
+			})
+
+			this.socket.on('error', (err) => {
+				this.updateStatus(InstanceStatus.ConnectionFailure, err.message)
+				this.log('error', 'Network error: ' + err.message)
+			})
+
+			this.socket.on('data', (data) => {
+				if (this.config.saveresponse) {
```

Comment:
There is no property `saveresponse` in your config. Please do proper adjustments when doing copy/paste job.

Reply:
its ok